mono/metadata/method-builder.h

```diff
@@ -21,22 +21,16 @@
 
 G_BEGIN_DECLS
 
-typedef struct _MonoMethodBuilder {
-	MonoMethod *method;
-	char *name;
-	gboolean no_dup_name;
-#ifdef ENABLE_ILGEN
-	GList *locals_list;
-	int locals;
-	gboolean dynamic;
-	gboolean skip_visibility, init_locals;
-	guint32 code_size, pos;
-	unsigned char *code;
-	int num_clauses;
-	MonoExceptionClause *clauses;
-	const char **param_names;
-#endif
-} MonoMethodBuilder;
+typedef struct _MonoMethodBuilder MonoMethodBuilder;
+
+#define MONO_METHOD_BUILDER_CALLBACKS_VERSION 1
+
+typedef struct {
```

Comment:
This looks like the wrong way to split MethodBuilder.

What if we spilt it along the lines of method header builder and method body builder?

This way we can link the whole body builder part into the extra lib.

Reply:
the body builders already live in `method-builder-ilgen.c`, but aren't actually exposed via the callback API. they are only needed by stuff that lives "behind the ILGEN wall".

does that make sense?